File: EXAMPLES/shmem_ptr_example.f90
PROGRAM REMOTEWRITE
INCLUDE 'shmem.fh'

INTEGER BIGD(100)
SAVE BIGD

INTEGER POINTEE(*)
POINTER (PTR,POINTEE)

CALL START_PES(0)


IF (MY_PE() .EQ. 0) THEN
   ! initialize PE 1's BIGD array
   PTR = SHMEM_PTR(BIGD, 1)     ! get address of PE 1's BIGD
                                !   array
   DO I=1,100
        POINTEE(I) = I
   ENDDO
ENDIF

CALL SHMEM_BARRIER_ALL

IF (MY_PE() .EQ. 1) THEN
   PRINT*,'BIGD on PE 1 is: '
   PRINT*,BIGD
ENDIF
END

File: EXAMPLES/shmem_ptr_example.c
#include <stdio.h>
#include <shmem.h>

int main(void) 
{
   static int bigd[100];
      int *ptr;
      int i;

   start_pes(0);

   if (_my_pe() == 0) {
      /* initialize PE 1's bigd array */
      ptr = shmem_ptr(bigd, 1);
      for (i=0; i<100; i++)
         *ptr++ = i+1;
   }

   shmem_barrier_all();

   if (_my_pe() == 1) {
      printf("bigd on PE 1 is:\n");
      for (i=0; i<100; i++)
         printf(" %d\n",bigd[i]);
      printf("\n");
   }
  return 1;
}
